### ISSUE TITLE
Add a transaction verification screen

### ISSUE THREAD

**Issue by Senyoret1** [skycoin/skycoin-web#148]
Please use https://github.com/skycoin/skycoin/pull/1408 as reference. The desktop wallet uses an api endpoint to obtain the transaction data, while the web wallet will have to build the transaction manually (at least for now), as it is doing it in `WalletService.sendSkycoin`.

**Comment by alexandermushinskiy** [comment]
@Senyoret1, just to be sure that we understood correctly: we should implement the same flow as in the desktop wallet: 
createTransaction() + injectTransaction() 
instead of 
sendSkycoin()
Is that right?

**Comment by Senyoret1** [comment]
Yes, the transaction must be created and shown in the preview component. Only after the user confirms the preview, the transaction is sent. Just like in the desktop wallet.

**Comment by Senyoret1** [comment]
It is not possible to use `wallet/transaction`, because the way the desktop wallet and the web wallet work is very different. The desktop wallet is really just a kind of control panel, since in fact who creates the wallets and performs all the processing is the node. When you create a new wallet in the desktop wallet, the desktop wallet ask the node to create it, and the node then creates a file on the hard disk with all the wallet data. The Id required by `wallet/transaction` is the name of the file created by the node, which does not exist when using the web wallet.

Due to this, the web wallet creates the transactions manually, instead of obtaining them by calling `wallet/transaction`. The process for creating transactions is currently in here:
https://github.com/skycoin/skycoin-web/blob/97ff6d21ce4fd1edfac26b044daed0f63bde6d62/src/app/services/wallet.service.ts#L70-L116

Most likely, it is necessary to modify that code so that it fully adapts to the needs of the preview.

**Comment by alexandermushinskiy** [comment]
@Senyoret1, thanks for the clarification.
One more question: there is the `<app-transaction-info>` component in the desktop wallet which is used to show transaction details. It is used in two pages (send verification and transaction list).
There is no page "transaction list" in the web wallet and my question is about the `<app-transaction-info>` component . Should it be generic as in desktop wallet (for future purpose) or can be created just for preview feature (send functionality)?

**Comment by Senyoret1** [comment]
The `TransactionListComponent` page from the desktop wallet is named `HistoryComponent` in the web wallet (I'm not sure why it has another name, but I guess it's not something that should be changed right now). That means `<app-transaction-info>` has to be generic, so it can be used on both pages. Also, note that in the desktop wallet `<app-transaction-info>` is not being used in `TransactionListComponent`, but in `TransactionDetailComponent`, which is the content of the modal screen that shows the details of the transactions in the history page.

**Comment by alexandermushinskiy** [comment]
@Senyoret1, the issue has been implemented, the PR is ready for review.

Too many changes in PR and we propose to create a separate issue for the following work:
- remove method walletService.sendSkycoin() due to it's not used anymore, remove appropriate unit-tets;
- add unit tests for new added methods: createTransaction, injectTransaction;
- add e2e tests for new flow in send sky coin;

Also if "transaction-info" component is going to be used in more then one place we recommend to create a separate folder "shared" and moved it there. What do you think?

**Comment by Senyoret1** [comment]
No problem, what remains to be done can be included in one or more PRs (as appropriate) after https://github.com/skycoin/skycoin-web/pull/156 is merged. As you say, the PR have many changes and adding more would be problematic.
